rclcpp/src/rclcpp/node.cpp

```diff
@@ -47,25 +47,28 @@ using rclcpp::Node;
 using rclcpp::NodeOptions;
 using rclcpp::exceptions::throw_from_rcl_error;
 
+namespace
+{
+
 RCLCPP_LOCAL
 std::string
 extend_sub_namespace(const std::string & existing_sub_namespace, const std::string & extension)
 {
   // Assumption is that the existing_sub_namespace does not need checking
   // because it would be checked already when it was set with this function.
 
-  // check if the new sub-namespace extension is absolute
-  if (extension.front() == '/') {
+  if (extension.empty()) {
```

Comment:
getting rid of `existing_sub_namespace.empty()` actually changes the behavior between former and current, but i think it should be okay.

Reply:
Yeah, that is true.  It changes it slightly, but I don't think in a way that most users will notice.

Previous to this PR there are 4 cases:

| | empty existing_sub_namespace | not empty existing namespace |
--- | --- | ---
| empty extension | throw exception |  new_subnamespace = existingnamespace (arguably buggy) |
| not empty extension | new subnamespace = extension | new subnamespace = existingnamespace + extension |

After this PR, the 4 cases look like:

| | empty existing_sub_namespace | not empty existing namespace |
--- | --- | ---
| empty extension | throw exception |  throw exception |
| not empty extension | new subnamespace = extension | new subnamespace = existingnamespace + extension |